tests/product/image_builder.py

```diff
@@ -0,0 +1,203 @@
+                #
+                # Once that's done, the cluster and test environment is in the
+                # same state it would be as if we'd called _run_installers on
+                # a bare cluster, and we can return.
+                #
+                # We do this to save the cost of running the installers on the
+                # docker containers every time we run a test. In practice,
+                # that turns out to be a fairly expensive thing to do.
+                if not bare_cluster:
+                    self._apply_post_install_hooks(installers)
+                    self._update_replacement_keywords(installers)
+                    return
+            except DockerClusterException as e:
+                print e.msg
+                exit(1)
+
+        # If we got a bare cluster back, we need to run the installers on it.
+        # applying the post-install hooks and updating the replacement
+        # keywords is handled internally in _run_installers.
+        self._run_installers(installers)
+
+        if isinstance(self.cluster, DockerCluster):
+            self.cluster.commit_images(bare_image_provider, cluster_type)
+
+    def _get_expected_image_names(self, cluster_type):
+        if cluster_type == self.BARE_CLUSTER:
+            slave_image_name = 'teradatalabs/pa_test/nohadoop_bare_slave'
+            master_image_name = 'teradatalabs/pa_test/nohadoop_bare_master'
+            return [slave_image_name, master_image_name]
+        elif cluster_type == self.PA_ONLY_CLUSTER:
+            slave_image_name = 'teradatalabs/pa_test/nohadoop_pa_only_standalone_slave'
+            master_image_name = 'teradatalabs/pa_test/nohadoop_pa_only_standalone_master'
+            return [slave_image_name, master_image_name]
+        elif cluster_type == self.STANDALONE_PRESTO_CLUSTER:
+            slave_image_name = 'teradatalabs/pa_test/nohadoop_presto_slave'
+            master_image_name = 'teradatalabs/pa_test/nohadoop_presto_master'
+            return [slave_image_name, master_image_name]
+        elif cluster_type == self.PA_ONLY_YS_CLUSTER:
+            slave_image_name = 'teradatalabs/pa_test/nohadoop_pa_only_ys_slave'
+            master_image_name = 'teradatalabs/pa_test/nohadoop_pa_only_ys_master'
+            return [slave_image_name, master_image_name]
+        else:
+            return []
+
+    def _assert_expected_images_created(self, cluster_type):
+        docker_images = check_output(['docker', 'images'])
+        expected_images = self._get_expected_image_names(cluster_type)
+        for expected_image in expected_images:
+            assert expected_image in docker_images
+
+    def _setup_cluster_with_no_hadoop_provider_assert_images(self, cluster_type):
+        self._setup_cluster(NoHadoopBareImageProvider(),
+                            cluster_type)
+        self._assert_expected_images_created(cluster_type)
+
+    def test_setup_standalone_presto_images(self):
```

Comment:
The `test` prefix for all these methods is no longer needed since they're no longer tests.